internal/proxy/oauthproxy.go

```diff
@@ -283,6 +291,25 @@ func NewOAuthProxy(opts *Options, optFuncs ...func(*OAuthProxy) error) (*OAuthPr
 		c.Run()
 	}()
 
+	// Configure the RequestSigner (used to sign requests with `Octoboi-Signature` header).
+	// Also build the `certs` static JSON-string which will be served from a public endpoint.
+	// The key published at this endpoint allows upstreams to decrypt the `Octoboi-Signature`
+	// header, and validate the integrity and authenticity of a request.
+	var certs = map[string]string{}
```

Comment:
Let's use a struct here instead of string map. We can use struct tags to make the json marshalling easy.

```golang
type Certificates struct {
    PublicKey string `json:signing_key`
}
. . .
certs := &Certificates{PublicKey:opts.RequestSigningKey}
. . .

rawBytes, err := json.MarshalIndent(*certs, "", "  ")
```

Reply:
Good idea.